.github/workflows/packaging.yml

```diff
@@ -0,0 +1,87 @@
+name: Packaging
+on:
+  push:
+    branches: 
+      - master
+      - devel
+      - 'release-**'
+jobs:
+  debian:
+    name: Debian package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install required packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y git-buildpackage build-essential dh-apache2 pkg-php-tools phpab php-xdebug
+          cd ~
+          # These need to be more recent than the ones shipped in ubuntu focal (to build the dependencies)
+          # We need these versions to build the dependencies. The previous ones don't work well with php 7.4 that is in ubuntu focal.
+          wget http://ftp.us.debian.org/debian/pool/main/p/php-zeta-console-tools/php-zeta-console-tools_1.7.2-2_all.deb
+          wget http://ftp.us.debian.org/debian/pool/main/p/phpab/phpab_1.26.0-1_all.deb
+          # There is this crash with dh_phpcomposer on ubuntu focal
+          # Non-static method Pkgtools\Base\Overrides::override() cannot be called statically in /usr/share/php/pkgtools/base/dependencies.php:45
+          # So we update pkg-php-tools to version 1.40 (focal has 1.38)
+          wget http://ftp.us.debian.org/debian/pool/main/p/pkg-php-tools/pkg-php-tools_1.40_all.deb
+          sudo dpkg -i php-zeta-console-tools_1.7.2-2_all.deb phpab_1.26.0-1_all.deb pkg-php-tools_1.40_all.deb
+      - name: Set env
+        run: |
+          echo "LAST_COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "LAST_COMMIT_DATE=$(git log -1 --date=format:%Y%m%d --format=%cd)" >> $GITHUB_ENV
+          echo "KALKUN_VERSION=$(grep kalkun_version application/config/kalkun_settings.php | cut -d "'" -f 4)" >> $GITHUB_ENV
+          echo "DEBEMAIL=none@domain.tld" >> $GITHUB_ENV
+          echo "DEBFULLNAME='Deb Builder'" >> $GITHUB_ENV
+      - name: Build kalkun packages
+        run: |
+          set -x
+          git remote add debian https://salsa.debian.org/bastif/kalkun.git
```

Comment:
Points to different upstream, maybe we should have another repo in github/ or have required files in this repo itself.

Reply:
This is the debian dev repo designed for all debian packages.
Doing it this way permits to provide sort of nightly package here.

Reply:
It would make more sense to keep all codes on the same `org`.  I would suggest that we move the debian support files either to this repo or to `https://github.com/kalkun-sms/kalkun-debian`, and keep all the code at github itself.

Reply:
I'm not fond of having to manage 2 repos for the same thing. The one I reference on salsa is mine. And it is the one where debian packages are hosted. So no, I don't think it is a good idea to have duplicate work for this.

Reply:
> I'm not fond of having to manage 2 repos for the same thing

But here we are still ending up with 2 repos, one on `github.com` and the other at `salsa.debian.org`. And the one at salsa is a personal repo, so if some other contributor wants to make some contributions it wouldn't be easy.

My suggestion is that either we have all code in github itself. We can have all in repo itself if that simpler.

Reply:
What I meant by 2 repos is one having a debian folder here, and one having a debian folder on salsa.

Debian salsa is the standard VCS system for debian packaging. Please understand that debian packaging has various constraints and specific tools (especially CI pipelines for packaging testing/lintian checks...). 
Debian packages can be either "native" debian packages or not. Most are not. Those that aren't (like kalkun) combine a "debian.tar.gz" and an "orig.tar.gz" (upstream source tarball  usually kept as is except removal of non DFSG compliant code). The orig.tar.gz shouldn't have a debian dir, while the debian.tar.gz has nothing except the debian dir and its contents.

I don't advise having a debian folder here. That would add complexity. Maintaining the debian folder, hence the packaging is a separate 'job'.

Reply:
I am not so much familiar with `dpkg-buildpackage`, I usually use `dpkg-deb` to package to .deb, I will try to read through the `dpkg-buildpackage` documentation to understand, but one question that I have is why does the repo at salsa have the kalkun code then? Shouldn't it just have the `debian` folder? That way that repo will not have outdated code ever.

Reply:
I'm rather new also to git & debian packages (see [git-buildpackage](https://manpages.debian.org/bullseye/git-buildpackage/git-buildpackage.1.en.html)) but there are upstream branches, and master branches. Orig only contains what is in upstream and is versioned based on what is released upstream, then there is the master branch which contains the debian folder (and the code pulled from  branches named upstream). It's rather new to me too so I certainly haven't understood all yet but all this works in a specific way to be able to use debian tools.

The ci pipelines on salsa are very valuable and replicate many checks that are performed by the official package builders.

Reply:
I also noticed that releasing a debian version of any package is not that easy. One has to check the license file for every release. Then update the patches, check new files that have to be installed or are not here anymore, do all the stuff that makes the package fit for a debian system. Kalkun is not easy to package at all with all its dependencies to external php libs that don't exist yet in debian.